### ISSUE TITLE
Can not use custom artifact zip in v1.18

### ISSUE THREAD

**Issue by sethkor** [serverless/serverless#4015]
This is a Bug Report 

Can not use custom artifacts in v1.18

When upgrading to v1.18 of the serverless framework, the artifact tag in the serveless.yml file is ignored.  Instead it searches for an artifact named with the service: tag in the yaml file.

For me I had this tag under:

```yml
functions:
   MyFunction:
   package:
      artifact: ./lambda.zip
```

Reverting back to serverless 1.16 and also testing with 1.17 i can see the behaviour I expect with my custom zip specified by the artifact tag being used for deployment.

I’m using nodejs 6.10 on AWS lambda.

**Comment by pmuens** [comment]
Thanks for opening @sethkor 👍

@eahefnawy could that be related to the recent fix with artifacts on a function level? 🤔

**Comment by HyperBrain** [comment]
@pmuens @eahefnawy I ran into the same issue with the serverless-webpack plugin. The move of the artifact property from `function.artifact` to `function.package.artifact` broke some functionality that depended on that.

Imo it would be better to have a `serverless.service.getFunctionArtifact(funcname)` and `serverless.service.setFunctionArtifact(funcname, artifactpath)` that would not break if the internal structures change.

A quick fix would be that the framework just sets and reads both variants (with the package.artifact being the preferred one).

In my case I used serverless.getVersion() to determine the correct format and write and read the props accordingly where I have to.

**Comment by HyperBrain** [comment]
BTW: It would be great, if SLS finally would switch to SemVer. Any breaking change should increase the _MAJOR_ number of the package, so that ppl can prepare themselves when switching. Having product management policies interfere with package versioning is a somehow contra-productive approach regarding customer satisfaction (in case of NPM packages and tools). The product itself can still be "v1" which is to be seen more virtual (PM side), but the package versions should increase according to semver.

**Comment by eahefnawy** [comment]
@sethkor @HyperBrain thanks for reporting and sorry for the delay on the response. We're super busy in the background preparing for our [Emit Conference](http://emitconference.com). But we'll def look into this and patch release it asap! 😊

**Comment by davidfox-ap** [comment]
Can anyone suggest a workaround, other than downgrading 1.17, while we wait for a patch?

**Comment by pmuens** [comment]
@davidfox-ap thanks for your comment 👍

What Serverless version are you using? Have you tried the most recent one?

**Comment by davidfox-ap** [comment]
I upgraded to 1.21 today (previously I was on 1.20.2, I believe) and still experienced the same issue. 

`No tag-59e21bac-8dc1-11e7-8b7b-62e9bfe7299a.zip file found in the package path you provided.`

'tag-59e21bac-8dc1-11e7-8b7b-62e9bfe7299a' is the name of my service. My artifact has a different name:

    package:
      artifact: tagger.zip

As an experiment, I changed the zip file name to match my service name and I still go the same error message. I was mystified, but this open issue seemed related to my own, so thought I would check in here to see if there was something I could do on my end to get things working with the latest version.
